aiida/orm/implementation/sqlalchemy/comments.py

```diff
@@ -109,11 +125,24 @@ def delete(self, comment):
         """
         # pylint: disable=no-name-in-module,import-error
         from sqlalchemy.orm.exc import NoResultFound
-        from aiida.backends.sqlalchemy import get_scoped_session
         session = get_scoped_session()
 
         try:
             session.query(models.DbComment).filter_by(id=comment).one().delete()
             session.commit()
         except NoResultFound:
             raise exceptions.NotExistent("Comment with id '{}' not found".format(comment))
+
+    # pylint: disable=no-self-use
+    def delete_many(self, filters):
+        """
+        Delete all comment entries in the table
+        """
+        session = get_scoped_session()
+
+        if not filters:
+            for entry in models.DbComment.query.all():
```

Comment:
This will be very likely inefficient. Maybe you can combine the deletion with the query
https://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.delete
Just check that we don't violate anything (written in the warnings - I think not from a first look)

Reply:
Deleting the entries one by one is the conventional way to go in sqlalchemy, deleting in bulk has caveats like not running cascade effects in python and possibly contaminating the state of the session. On the other hand, bulk delete is actually more performant, but I'd say those performance gains become important only when deleting a huge amount of objects.

Reply:
Reading the docs referenced by @szoupanos I understand that the cascading is still observed as long as it is defined, when the column containing the ForeignKey is defined. This should be the true for all relevant cases, meaning this should be fine.
On another note, I can see that I forgot to do a try, escape for Comments (only did it for Logs).

Reply:
Additionally, for Comments the associated Node or User should never be deleted (however a Comment may be deleted, when deleting a Node).
I will delete in bulk then, using `session.query(models.DbComment).delete()` and similarly for Logs.